test/queries/ledger_test.go

```diff
@@ -633,3 +634,176 @@ func verifyTxProof(intermediateHashes [][]byte, receipt *types.TxReceipt, tx pro
 
 	return bytes.Equal(receipt.GetHeader().GetTxMerkelTreeRootHash(), currHash), nil
 }
+
+// Scenario:
+// HTTP GET "/ledger/proof/data/{blockId}/{dbname}/{key}?deleted={true|false}" gets proof for value associated with (dbname, key) in block blockId,
+// HTTP GET "/ledger/proof/data/{blockId}/{dbname}/{key}" gets proof for value associated with (dbname, key) in block blockId
+func TestLedgerDataProof(t *testing.T) {
```

Comment:
This looks good, but minor comments. 

Please add a test case that does the "positive" cases with async txs, i.e. block with more than one tx, like we did for the tx-proof.